lib/omise/OmiseCapabilities.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+class OmiseCapabilities extends OmiseApiResource
+{
+    const ENDPOINT = 'capability';
+
+    const FILTERS = array(
+        'backend' => array('currency', 'type', 'chargeAmount')
+    );
+
+    protected function __construct($publickey = null, $secretkey = null)
+    {
+        parent::__construct($publickey, $secretkey);
+        $this->_setupFilterShortcuts();
+    }
+
+    /**
+     * Sets up 'shortcuts' to filters so they may be used thus:
+     *    $capabilities->backendFilter['currency']('THB')
+     * As well as the original:
+     *    $capabilities->makeBackendFilterCurrency('THB')
+     */
+    protected function _setupFilterShortcuts()
+    {
+        foreach (self::FILTERS as $filterSubject=>$availableFilters) {
+            $filterArrayName = $filterSubject.'Filter';
+            $this->$filterArrayName = array();
+            $tempArr = &$this->$filterArrayName;
+            foreach ($availableFilters as $type) {
+                $funcName = "make".ucfirst($filterSubject).'Filter'.$type;
+                $tempArr[$type] = function() use ($funcName) { return call_user_func_array(array($this, $funcName), func_get_args()); };
+            }
+        }
+    }
+
+    /**
+     * Retrieves capabilities.
+     *
+     * @param  string $publickey
+     * @param  string $secretkey
+     *
+     * @return OmiseCapabilities
+     */
+    public static function retrieve($publickey = null, $secretkey = null)
+    {
+        return parent::g_retrieve(get_class(), self::getUrl(), $publickey, $secretkey);
+    }
+
+    /**
+     * Retrieves array of payment backends. Optionally pass in as many filter functions as you want
+     * (muliple arguments, or a single array)
+     *
+     * @param [func1,fun2,...] OR func1, func2,...
+     *    
+     * @return array
+     */
+    public function getBackends()
+    {
+        // check for filters
+        if ($filters = func_get_args()) $filter = self::_combineFilters(self::_argsToVariadic($filters));
+        $res = $this['payment_backends'];
+        array_walk(
+            $res,
+            function($v, $k) use (&$res) {
+                $id = array_keys($v)[0];
+                $res[$k][$id]['_id'] = $id;
+            }
+        );
+        $res = array_map(function($a) { return (object)reset($a); }, $res);
```

Comment:
@jonrandy I'm combining these `array_walk` and `array_map` to 
```php
$backends = array_map(
    function($backend) {
        $id = array_keys($backend)[0];
        $backend[$id]['_id'] = $id;
        return (object)reset($backend);
    }, 
    $backends);

return !empty($filter) ? array_values(array_filter($backends, $filter)) : $backends;
```

Do you have any concern from the above code? (I tested, the result seems to be the same)

Reply:
As long as it gives the same result. I don't see that the `array_values` call is needed though - who cares about the indexes? Wasteful additional processing